src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs

```diff
@@ -412,11 +414,16 @@ private PropertyDescriptorCollection GetProperties()
             return propertyDescriptors;
         }
 
+        [RequiresUnreferencedCode("PropertyDescriptor's PropertyType cannot be statically discovered.")]
         protected virtual void GetProperties(Hashtable propertyDescriptors)
         {
             long logScopeId = DataCommonEventSource.Log.EnterScope("<comm.DbConnectionStringBuilder.GetProperties|API> {0}", ObjectID);
             try
             {
+                // Below call is necessary to tell the trimmer that it should mark derived types appropriately.
+                // We cannot use overload which takes type because the result might differ if derived class implements ICustomTypeDescriptor.
+                Type thisType = GetType();
```

Comment:
I would remove this. The method is `RequiresUnreferencedCode`, we don't make any guarantees that things are going to be preserved if you call it.

Reply:
I think it might still be a good idea to at least try to make it work as much as we can even though there is no guarantees

Reply:
(or rather doesn't hurt to have it working for some scenarios)